imagetest/testworkflow.go

```diff
@@ -225,6 +254,118 @@ func (t *TestWorkflow) addStartStep(stepname, vmname string) (*daisy.Step, error
 	return startInstancesStep, nil
 }
 
+func (t *TestWorkflow) addCreateNetworkStep(networkName string) (*daisy.Step, error) {
+	createNetworks := &daisy.CreateNetworks{}
+	pFalse := false
+	network := &daisy.Network{
+		Network: compute.Network{
+			Name: networkName,
+		},
+		AutoCreateSubnetworks: &pFalse,
+	}
+	*createNetworks = append(*createNetworks, network)
+
+	createNetworksStep, err := t.wf.NewStep(createNetworkStepName)
+	if err != nil {
+		return nil, err
+	}
+	createNetworksStep.CreateNetworks = createNetworks
+
+	return createNetworksStep, nil
+}
+
+func (t *TestWorkflow) addCreateSubnetworkStep(vmname, networkName, subnetworkName, rangeName, primaryRange, secondaryRange string) (*daisy.Step, error) {
+	createSubnetworks := &daisy.CreateSubnetworks{}
+	subnetwork := &daisy.Subnetwork{
+		Subnetwork: compute.Subnetwork{
+			Name:        subnetworkName,
+			IpCidrRange: primaryRange,
+			Network:     networkName,
+			SecondaryIpRanges: []*compute.SubnetworkSecondaryRange{
+				{
+					IpCidrRange: secondaryRange,
+					RangeName:   rangeName,
+				},
+			},
+		},
+	}
+	*createSubnetworks = append(*createSubnetworks, subnetwork)
+
+	createSubnetworksStep, err := t.wf.NewStep("create-sub-network-" + vmname)
+	if err != nil {
+		return nil, err
+	}
+	createSubnetworksStep.CreateSubnetworks = createSubnetworks
+
+	return createSubnetworksStep, nil
+}
+
+// CreateSubNetwork creates custom subnetwork. Using SetCustomNetwork method
+// provided by TestVM to config network on vm
+func (n Network) CreateSubNetwork(name string, ipRange string) (*SubNetwork, error) {
+	createSubnetworks := &daisy.CreateSubnetworks{}
+	subnetwork := &daisy.Subnetwork{
+		Subnetwork: compute.Subnetwork{
+			Name:        name,
+			IpCidrRange: ipRange,
+			Network:     n.Name,
+		},
+	}
+	*createSubnetworks = append(*createSubnetworks, subnetwork)
+
+	createSubnetworksStep, err := n.testWorkflow.wf.NewStep(createSubNetworkStepName)
```

Comment:
i think we always want to use the one create-subnets step  (you created a constant for the name), instead of creating new steps each time

Reply:
you do need to create the step the first time, if it doesnt exist. otherwise find and append to it, like we do with create-vms